src/PrestaShopBundle/Form/Admin/Sell/Product/Combination/BulkCombinationStockType.php

```diff
@@ -0,0 +1,157 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Sell\Product\Combination;
+
+use PrestaShopBundle\Form\Admin\Type\DatePickerType;
+use PrestaShopBundle\Form\Admin\Type\DeltaQuantityType;
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Type;
+
+class BulkCombinationStockType extends TranslatorAwareType
+{
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var bool
+     */
+    private $stockManagementEnabled;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param RouterInterface $router
+     * @param bool $stockManagementEnabled
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        RouterInterface $router,
+        bool $stockManagementEnabled
+    ) {
+        parent::__construct($translator, $locales);
+        $this->router = $router;
+        $this->stockManagementEnabled = $stockManagementEnabled;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ($this->stockManagementEnabled) {
+            $builder
+                ->add('delta_quantity', DeltaQuantityType::class, [
+                    'required' => false,
+                    'label' => $this->trans('Edit quantity', 'Admin.Catalog.Feature'),
+                    'disabling_switch' => true,
+                    'disabled_value' => function (?array $data) {
+                        return empty($data['quantity']) && empty($data['delta']);
```

Comment:
Does `empty($data['quantity'])` means that a "0 quantity" stock can't apply a "delta" ?

Reply:
No, actually the `disabled_value` is only used on the first form rendering to detect if it's considered in its disabled state or not For most cases just indicating a disabled value is enough

Ex: `'disabled_value' => -1` => when the field has a value of `-1` it is considered disabled

But in this particular case the option is set on a compound form which is composed of multiple fields so we can't set a simple value

On such cases we use a callback which is able to analyze both fields. So here `delta_quantity` is disabled when `quantity` AND `delta` is zero, meaning there is no modification basically